front/components/data_source/DataSourceEdition.tsx

```diff
@@ -0,0 +1,436 @@
+              ? {
+                  ...ds,
+                  connector: createdManagedDataSource.connector,
+                  setupWithSuffix: null,
+                  dataSourceName: createdManagedDataSource.dataSource.name,
+                }
+              : ds;
+          })
+        );
+        if (REDIRECT_TO_EDIT_PERMISSIONS.includes(provider)) {
+          void router.push(
+            `/w/${owner.sId}/builder/data-sources/${createdManagedDataSource.dataSource.name}?edit_permissions=true`
+          );
+        }
+      } else {
+        const responseText = await res.text();
+        sendNotification({
+          type: "error",
+          title: `Failed to enable connection (${provider})`,
+          description: `Got: ${responseText}`,
+        });
+      }
+    } catch (e) {
+      onClose();
+      sendNotification({
+        type: "error",
+        title: `Failed to enable connection (${dataSourceIntegration.connectorProvider})`,
+      });
+    } finally {
+      setIsLoadingByProvider((prev) => ({
+        ...prev,
+        [dataSourceIntegration.connectorProvider]: false,
+      }));
+    }
+  };
+
+  const updateConnectorConnectionId = async (
+    newConnectionId: string,
+    provider: string
+  ) => {
+    const res = await fetch(
+      `/api/w/${owner.sId}/data_sources/${dataSourceName}/managed/update`,
+      {
+        method: "POST",
+        headers: {
+          "Content-Type": "application/json",
+        },
+        body: JSON.stringify({
+          connectionId: newConnectionId,
+        } satisfies UpdateConnectorRequestBody),
+      }
+    );
+
+    if (res.ok) {
+      return { success: true, error: null };
+    }
+
+    const jsonErr = await res.json();
+    const error = jsonErr.error;
+
+    if (error.type === "connector_oauth_target_mismatch") {
+      return {
+        success: false,
+        error: CONNECTOR_TYPE_TO_MISMATCH_ERROR[provider as ConnectorProvider],
+      };
+    }
+    return {
+      success: false,
+      error: `Failed to update the permissions of the Data Source: (contact support@dust.tt for assistance)`,
+    };
+  };
+
+  const handleUpdatePermissions = async () => {
+    if (!dataSourceIntegration) {
+      return;
+    }
+
+    const connectionIdRes = await setupConnection({
+      dustClientFacingUrl,
+      owner,
+      provider: connectorProvider,
+    });
+    if (connectionIdRes.isErr()) {
+      sendNotification({
+        type: "error",
+        title: "Failed to update the permissions of the Data Source",
+        description: connectionIdRes.error.message,
+      });
+      return;
+    }
+
+    const updateRes = await updateConnectorConnectionId(
+      connectionIdRes.value,
+      connectorProvider
+    );
+    if (updateRes.error) {
+      sendNotification({
+        type: "error",
+        title: "Failed to update the permissions of the Data Source",
+        description: updateRes.error,
+      });
+    }
+  };
+  return (
```

Comment:
I feel like most of this code is not used, at least I can't see it when browsing the app locally. Is it some legacy behavior where you thought that this was the entry point to setup new connections?

Reply:
Well this code is triggered when clicking the "Edit Permissions" button ?